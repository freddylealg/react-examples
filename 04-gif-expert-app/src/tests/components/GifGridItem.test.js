import React from 'react';
import { shallow } from 'enzyme'
import '@testing-library/jest-dom';
import toJson from 'enzyme-to-json';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas del componente GifGridItem', () => {

    const title = 'título';
    const url = 'url';
    let wrapper = shallow( <GifGridItem title={title} url={url} /> );

    beforeEach( () => { 
        wrapper = shallow( <GifGridItem title={title} url={url} /> );
    });

    test('Provar visualización del componente', () => {
        expect( toJson(wrapper) ).toMatchSnapshot();
    });

    test('Debe de tener un parrafo con el título', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title )
    });

    test('Debe tener la imagen de la url de los props', () => {
        const img = wrapper.find('img');
        expect( img.prop('src').trim() ).toBe( url )
        expect( img.prop('alt').trim() ).toBe( title )
    });

    test('Debe tener la clase animate__fadeIn', () => {
        const div = wrapper.find('div');
        expect( div.prop('className').includes('animate__fadeIn') ).toBe( true );
    });

});