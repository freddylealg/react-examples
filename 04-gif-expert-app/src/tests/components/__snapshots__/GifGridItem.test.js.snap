// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifGridItem Provar visualización del componente 1`] = `
<div
  className="card animate__animated animate__bounce animate__fadeIn"
>
  <img
    alt="título"
    src="url"
  />
  <p>
    título
  </p>
</div>
`;
